modules/mobile-agents/android/client/plugins/IDPProxy/src/org/wso2/mdm/agent/proxy/Token.java

```diff
@@ -0,0 +1,92 @@
+package org.wso2.mdm.agent.proxy;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+
+/**
+ * Persists refresh token to obtain new access token and id token to retrieve
+ * login user claims
+ */
+public final class Token {
+	private String refreshToken = null;
+	private String idToken = null;
+	private String accessToken = null;
+	private Date receivedDate = null;
+	@SuppressWarnings("unused")
+	private boolean expired = false;
+
+	public Date getDate() {
+		return receivedDate;
+	}
+
+	public void setDate() {
+		DateFormat dateFormat = new SimpleDateFormat("MM/dd/yyyy HH:mm:ss", Locale.getDefault());
```

Comment:
Let's please do this initialization off the critical path of a request. In other words, initialize this when the class is loaded and re-use it for all downstream code.